math-scala/src/main/scala/org/apache/mahout/math/algorithms/transformer/AsFactor.scala

```
@@ -0,0 +1,79 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one
+  * or more contributor license agreements. See the NOTICE file
+  * distributed with this work for additional information
+  * regarding copyright ownership. The ASF licenses this file
+  * to you under the Apache License, Version 2.0 (the
+  * "License"); you may not use this file except in compliance
+  * with the License. You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing,
+  * software distributed under the License is distributed on an
+  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  * KIND, either express or implied. See the License for the
+  * specific language governing permissions and limitations
+  * under the License.
+  */
+
+package org.apache.mahout.math.algorithms.transformer
+
+import org.apache.mahout.math.{Vector => MahoutVector}
+
+import collection._
+import JavaConversions._
+import org.apache.mahout.math._
+import org.apache.mahout.math.drm._
+import org.apache.mahout.math.drm.RLikeDrmOps._
+import org.apache.mahout.math.scalabindings._
+import org.apache.mahout.math.scalabindings.RLikeOps._
+import scala.reflect.ClassTag
+
+class AsFactor extends Transformer{
+
+  var factorMap: MahoutVector = _
+  var k: MahoutVector = _
+  var summary = ""
+
+  def transform[K: ClassTag](input: DrmLike[K]): DrmLike[K] ={
+    if (!isFit) {
+      //throw an error
```

Is this complete? I.e., do you want to throw an error here for this release?